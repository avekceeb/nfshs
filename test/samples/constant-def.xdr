/*

RFC 4506

6.2.4
A decimal constant expresses a number in base 10 and is a sequence of one or more decimal digits,
where the first digit is not a zero, and is optionally preceded by a minus-sign (’-’).

6.2.5
A hexadecimal constant expresses a number in base 16, and must be preceded by ’0x’,
followed by one or more hexadecimal digits (’A’, ’B’, ’C’, ’D’, E’, ’F’, ’a’, ’b’, ’c’, ’d’, ’e’, ’f’,
’0’, ’1’, ’2’, ’3’, ’4’, ’5’, ’6’, ’7’, ’8’, ’9’).

6.2.6
An octal constant expresses a number in base 8, always leads with digit 0,
and is a sequence of one or more octal digits (’0’, ’1’, ’2’, ’3’, ’4’, ’5’, ’6’, ’7’).

*/

const DEC  = 42;
const DECN = -0;

const HEX = 0xCAFE42babe;

const OCT = 001234567;
