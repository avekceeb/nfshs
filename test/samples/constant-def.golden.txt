( [ Right (ConstantDef (Identifier "DEC") (DecConstant 42))
  , Right (ConstantDef (Identifier "DECN") (DecConstant 0))
  , Right (ConstantDef (Identifier "HEX") (HexConstant 871849179838))
  , Right (ConstantDef (Identifier "OCT") (OctConstant 342391))
  ]
, ParserState
    [ ( SourcePos
          { sourceName = "test/samples/constant-def.xdr"
          , sourceLine = Pos 25
          , sourceColumn = Pos 1
          }
      , ConstantDef (Identifier "OCT") (OctConstant 342391)
      )
    , ( SourcePos
          { sourceName = "test/samples/constant-def.xdr"
          , sourceLine = Pos 23
          , sourceColumn = Pos 1
          }
      , ConstantDef (Identifier "HEX") (HexConstant 871849179838)
      )
    , ( SourcePos
          { sourceName = "test/samples/constant-def.xdr"
          , sourceLine = Pos 21
          , sourceColumn = Pos 1
          }
      , ConstantDef (Identifier "DECN") (DecConstant 0)
      )
    , ( SourcePos
          { sourceName = "test/samples/constant-def.xdr"
          , sourceLine = Pos 20
          , sourceColumn = Pos 1
          }
      , ConstantDef (Identifier "DEC") (DecConstant 42)
      )
    ]
    []
    (fromList [] :| [])
)